app/lib/user.server.ts

```diff
@@ -4,5 +4,5 @@ import { deleteUserByUsername } from '~/models/user.server';
 
 export async function deleteUser(username: User['username']) {
   await deleteUserByUsername(username);
-  setIsReconciliationNeeded(true);
+  return setIsReconciliationNeeded(true);
```

Comment:
We could do a follow-up to audit all callers of `setIsReconciliationNeeded(true)` to see if we can move them into the model methods per @cychu42's comment, if you don't want to do it here.